frontend/src/components/RecordsList.tsx

```diff
@@ -27,10 +28,14 @@ export interface IRecordsListProps {
   pageRoute: string;
   searchTerm: string;
   colDefs: ColDef[];
-  conditionBuilder: (val: string) => Record<string, any>[];
+  conditionBuilder: (uniqueQueries: string[]) => Record<string, any>[];
   sampleQueryParamValue: string | undefined;
   sampleQueryParamFieldName: string;
   searchVariables: SampleWhere;
+  customFilterUI?: JSX.Element;
+  customFilterState?: boolean;
+  setCustomFilterVals?: (vals: any[]) => void;
+  customFilterFunc?: (vals: any[]) => Promise<string[]>;
```

Comment:
These new props allow us to agnostically pass in the MRN search toggle widget and related functionalities. See `frontend/src/pages/patients/PatientsPage.tsx` to see how these props are being used.